pkg/metrics/metrics.go

```diff
@@ -0,0 +1,183 @@
+package metrics
+
+import (
+	"encoding/json"
+	"strings"
+
+	corelisters "k8s.io/client-go/listers/core/v1"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	configv1 "github.com/openshift/cluster-samples-operator/pkg/apis/samples/v1"
+	"github.com/openshift/cluster-samples-operator/pkg/client"
+	sampoplisters "github.com/openshift/cluster-samples-operator/pkg/generated/listers/samples/v1"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	sampleNamespace = "openshift_samples"
+	statusSubsystem = "status"
+
+	// per https://prometheus.io/docs/practices/naming/#metric-names feel our metrics here fall under the _info category of classification
+
+	// since we want to index by imagestream name, where each imagestream is either failing or not failing, this is why
+	// we went with the _info suffix vs. the _total suffix
+	failedImportsQuery = "openshift_samples_failed_imagestream_import_info"
+
+	degraded      = "degraded_info"
+	degradedQuery = "openshift_samples_status_degraded_info"
+
+	invalidConfig      = "invalidconfig_info"
+	invalidConfigQuery = "openshift_samples_status_invalidconfig_info"
+	invalidSecret      = "invalidsecret_info"
+	invalidSecretQuery = "openshift_samples_status_invalidsecret_info"
+
+	missingSecret        = "missing-secret"
+	missingTBRCredential = "missing-tbr-credential"
+
+	// MetricsPort is the IP port supplied to the HTTP server used for prometheus,
+	// and matches what is specified in the corresponding Service and ServiceMonitor
+	MetricsPort = 60000
+)
+
+var (
+	importsFailedDesc = prometheus.NewDesc(failedImportsQuery,
+		"Indicates by name whether a samples imagestream import has failed or not (1 == failure, 0 == succes).",
+		[]string{"name"},
+		nil)
+	invalidSecretDesc = prometheus.NewDesc(invalidSecretQuery,
+		"Indicates if the pull secret is functional in the openshift namespace.  If not functional, the reason supplied either points to it missing in the openshift namesapce, or if it is missing credentials for registry.redhat.io when they should exist (1 == not functional, 0 == functional).",
+		[]string{"reason"},
+		nil)
+	degradedStat = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: sampleNamespace,
```

Comment:
namespace and subsystem are likely to be removed in the future from the prometheus client, I recommend to just use `Name`

Reply:
removed use of namespace/subsystem